rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,602 @@
+    It runs first time through handle_button() when the start button clicked.
+    """
+    uid=globals()['ipy_wgs'][model_id]
+    
+    if globals()['rdkit_3dviewer_start_button_clicked_'+uid]:
+        
+        view = globals()['view_'+uid]
+        view.removeAllModels()
+        view.removeAllSurfaces()
+        view.removeAllLabels()
+        
+        uid = globals()['ipy_wgs'][model_id]
+        molViewState  = globals()['mol_views'][uid]
+        
+        
+        # handling mol and conf
+        molViewState = handleMolAndConf(molViewState)
+        
+        # handling properties
+        handleProperty(molViewState)
+        
+        # Add models (molecules/conformations) to viewer
+        pModelNum = 0
+        for model in molViewState.selectedModels:
+            pModelNum = pModelNum+1
+            globals()['view_'+uid].addModel(model, 'sdf')
+            
+        # Ligand Drawing style and colors
+        view=handleLigandStyle(uid, view)
+        # Ligand labeling
+        view = handleLigandLabeling(molViewState, view)
+        
+        # Add model (protein) to viewer
+        if 'protein_'+uid in globals():
+            # Show protein if visibitity is true
+            if globals()['proteinVisible_'+uid].value:
+                pdb = Chem.MolToPDBBlock(globals()['protein_'+uid])
+                view.addModel(pdb,'pdb')
+                # Protein Drawing style and colors
+                view = handleProteinStyle(uid, view, pModelNum)
+                
+        # zoomTo does not work well for surface and label... so, zoomTo should not be default settings
+        #view.zoomTo()
+        view.setBackgroundColor(globals()['background_'+uid].value)
+        
+        display(view.update())
+        
+    
+def handle_change(change):
+    """This function handles all the interactive widgets except buttons and 3Dmol.js viewer"""
+    update3D(change.owner._model_id)
+    
+def handle_start_button(b):
+    """This function handles start button."""
+    b.icon='check'
+    b.description="Done!"
+    uid=globals()['ipy_wgs'][b._model_id]
+    globals()['rdkit_3dviewer_start_button_clicked_'+uid] = True
+    update3D(b._model_id)
+    
+def handle_zoomTo_button(b):
+    """This function handles zoomTo button"""
+    uid=globals()['ipy_wgs'][b._model_id]
+    globals()['view_'+uid].zoomTo()
+    display(globals()['view_'+uid].update())
+    
+    
+def ChangeActiveLigand(uid, molId, confId, keepExistingModels = True):
+    """This function handles ligand select through python code
+    uid : string 
+    molId : string 
+    confId : intiger
+    keepExistingModels : logical (if False - all models are removed - molId and confId associated model rendered)
+    """
+    if type(uid) != 'str':
+        uid = str(uid)
+    if type(molId) != 'str':
+        molId = str(molId)
+        
+    if globals()['molId_'+uid].value != molId:
+        globals()['molId_'+uid].value = molId
```

Comment:
Considering your all the comments I am not sure whether the following lines will be required but writing those for the clarification purposes. Many thanks for commenting.
**Here is a note:** 
The lines mentioned is a part of `ChangeActiveLigand` function. This function actually make changes in widget's state and call the `update3D` function at the end. If I am not wrong, Greg wants to have a python function that can change models within an existing viewer. 
Now, for example, if conformer 1 of molecule m0 is already active in the viewer and if the user wants to see the conformer 2 of the same molecule then the function fails showing error since the widget for molecule selection got nothing to update.